ocp_resources/multi_network_policy.py

```diff
@@ -0,0 +1,57 @@
+from ocp_resources.resource import NamespacedResource
+
+
+class MultiNetworkPolicy(NamespacedResource):
+    """
+    MultiNetworkPolicy object.
+
+    API reference:
+    https://docs.openshift.com/container-platform/4.14/networking/multiple_networks/configuring-multi-network-policy.html
+    """
+
+    api_group = NamespacedResource.ApiGroup.K8S_CNI_CNCF_IO
+
+    def __init__(
+        self,
+        network_name=None,
+        policy_types=None,
+        ingress=None,
+        egress=None,
+        pod_selector=None,
+        **kwargs,
+    ):
+        """
+        Create and manage MultiNetworkPolicy
+
+        Args:
+            network_name (str): The name of the NetworkAttachmentDefinition that the policy will impact.
+            policy_types (list): One or both of the valid ip policies - "Ingress" and "Egress".
+            ingress (list): list containing a dictionary specifying the allowed "from" parameters.
+            egress (list): list containing a dictionary specifying the allowed "to" parameters.
+            pod_selector (dict): Map a label to match.
+        """
+        super().__init__(**kwargs)
+        self.network_name = network_name
+        self.policy_types = policy_types
+        self.pod_selector = pod_selector
+        self.ingress = ingress
+        self.egress = egress
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            if not self.network_name:
+                raise ValueError("network_name is required")
+
+            self.res["metadata"]["annotations"] = {
+                f"{self.api_group}/policy-for": f"{self.namespace} /{self.network_name}"
+            }
+            self.res["spec"] = {}
+            self.res["spec"].update({
+                "podSelector": {"matchLabels": self.pod_selector} if self.pod_selector else {},
+                "policyTypes": self.policy_types,
+            })
+            if self.ingress:
+                self.res["spec"].update({"ingress": self.ingress})
+            else:  # self.egress
```

Comment:
1. please remove the inline comment
2. see comment on empty list above
3. if one of them is mandatory, please add a check. if not, please add a separate condition - according to the doc, one of them must exist

Reply:
It is possible to create a policy without self.ingress or self.egress:
apiVersion: k8s.cni.cncf.io/v1beta1
kind: MultiNetworkPolicy
metadata:
  annotations:
    k8s.cni.cncf.io/policy-for: flat-overlay-ns/flat-l2-nad
  name: flat-overlay-mnp
  namespace: flat-overlay-ns
spec:
  podSelector:
    matchLabels:
      kubevirt.io/vm: vma
  policyTypes:
  - Egress
 
Can you please refer me to where in the documentation you saw that including at least one of them is mandatory?

I added a condition instead of the 'else'.

Reply:
> If you tested it, and it's working, then I guess t is OK.
> Did you test it?
I can't respond to your original message so I'm responding here - I did test it and the policy is created successfully. If the documentation is specifically claiming otherwise I want to make sure this is not a bug.